### PR TITLE
DOC: create a table for period aliases

### PR DESCRIPTION
**Pull request by natmokval**
related to pr #53477
Updated documentation for [Time series / date functionality](https://pandas.pydata.org/docs/user_guide/timeseries.html#time-series-date-functionality)

Created a new table for `period aliases`. Added to the doc of `Period dtypes`reference to the table `period aliases` instead of [offset aliases](https://pandas.pydata.org/docs/user_guide/timeseries.html#timeseries-offset-aliases). Corrected docs for `to_period`. Pointed out that only frequencies from the table `period aliases` are valid for `to_period`.

### CODE REVIEW

**Comment by natmokval** [comment]
@MarcoGorelli,  could you please take a look at this pr?

**Review by MarcoGorelli — COMMENTED** [review]
Nice table! Great to have this, thanks, this will help users

I just have a comment - you write "we will refer to these aliases as *period aliases*", but then in the `to_period` docstring you write "period strings". Perhaps that should say "period aliases" instead?

cc @jbrockmendel in case you have any comments

**Comment by natmokval** [comment]
> I just have a comment - you write "we will refer to these aliases as _period aliases_", but then in the `to_period` docstring you write "period strings". Perhaps that should say "period aliases" instead?
>

Thank you, @MarcoGorelli for the comment. I corrected `to_period` docstring, as you suggested.

File: doc/source/user_guide/timeseries.rst [hunk 1]
```diff
@@ -1299,6 +1299,31 @@ frequencies. We will refer to these aliases as *offset aliases*.
...
+
+.. csv-table::
+    :header: "Alias", "Description"
+    :widths: 15, 100
+
+    "B", "business day frequency"
+    "D", "calendar day frequency"
+    "W", "weekly frequency"
+    "M", "monthly frequency"
+    "Q", "quarter end frequency"
```
Comment (MarcoGorelli) [inline doc/source/user_guide/timeseries.rst:1318]: Wait, for period there is no "end", should this be "quarter frequency"?

Comment (natmokval) [reply doc/source/user_guide/timeseries.rst:1318]: Oops, my mistake. I'll correct it.

**Review by MarcoGorelli — APPROVED** [review]
Nice, looks good to me!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @natmokval

**Pull request merged**